### ISSUE TITLE
when the password have the @ char, have error

### ISSUE THREAD

**Issue by opspy** [uken/fluent-plugin-elasticsearch#283]
mypassword is `4xOofY@a`
```
2017-08-30 17:42:02 +0800 [warn]: temporarily failed to flush the buffer. next_retry=2017-08-30 17:42:32 +0800 error_class="URI::InvalidURIError" error="bad URI(is not URI?): http://elastic:4xOofY@a@172.19.1.2:9200/" plugin_id="object:3fc7c9a19b88"
```

**Comment by cosmo0920** [comment]
Could you use `%40` instead of `@` in your fluentd conf?

**Comment by cosmo0920** [comment]
```ruby
require 'uri'
URI.encode_www_form_component("@") # => "%40"
```

**Comment by opspy** [comment]
thanks

**Comment by cosmo0920** [comment]
I'm closing this issue.
If there is still any issue, feel free to reopen this.

**Comment by cosmo0920** [comment]
Ah..., Logstash ES plugin can handle `@`.
We should also handle it.

**Comment by cosmo0920** [comment]
I tried to implement username and password escaping feature in https://github.com/uken/fluent-plugin-elasticsearch/pull/306.
Does anyone want to try these patches?

**Comment by cosmo0920** [comment]
Released escape placeholders support in 2.0.0.rc.6 and 1.10.1.
If you noticed an undesired thing and found a bug, please reopen this issue.
